dashboard/controller/assignments/editAssignment.js

```diff
@@ -0,0 +1,100 @@
+//include libraries
+var superagent = require('superagent'),
+    moment = require('moment'),
+    common = require('../../helper/common'),
+    xocolors = require('../../helper/xocolors')(),
+    emoji = require('../../public/js/emoji'),
+    dashboard_utils = require('../dashboard/util');
+
+var assignments = require('./index');
+
+module.exports = function editAssignment(req, res) {
+    // reinit l10n and momemt with locale
+    common.reinitLocale(req);
+    if (req.params.assignmentId) {
+        if (req.method == 'POST') {
+            // validate
+            req.body.name = req.body.name.trim();
+            req.body.classrooms = req.body.classrooms || [];
+            if (typeof req.body.classrooms == 'string') {
+                req.body.classrooms = [req.body.classrooms];
+            }
+            //join dueDate and time
+            if (req.body.dueDate && req.body.time) {
+                req.body.dueDate = req.body.dueDate + " " + req.body.time;
```

Comment:
Convert datetime to unix timestamp as discussed.

Reply:
work in progress with dates